ui/store/action-queue/index.test.js

```diff
@@ -0,0 +1,196 @@
+      //   await submitRequestToBackground('backgroundFunction');
+      // }).toThrow('test');
+    });
+  });
+
+  describe('callBackgroundMethod', () => {
+    afterEach(() => {
+      sinon.restore();
+    });
+
+    it('calls background method if the stream is connected', () => {
+      const background = {
+        connectionStream: {
+          readable: true,
+        },
+        backgroundFunction: sinon.stub(),
+      };
+
+      _setBackgroundConnection(background);
+      callBackgroundMethod('backgroundFunction', [], () => ({}));
+      expect(background.backgroundFunction.called).toStrictEqual(true);
+    });
+
+    it('does not calls background method if the stream is not onnected', () => {
+      const background = {
+        connectionStream: {
+          readable: false,
+        },
+        backgroundFunction: sinon.stub(),
+      };
+
+      _setBackgroundConnection(background);
+      callBackgroundMethod('backgroundFunction', [], () => ({}));
+      expect(background.backgroundFunction.called).toStrictEqual(false);
+    });
+
+    it('calls background method on stream reconnection', () => {
+      const background = {
+        connectionStream: {
+          readable: false,
+        },
+        backgroundFunction: sinon.stub(),
+      };
+      _setBackgroundConnection(background);
+      callBackgroundMethod('backgroundFunction', [], () => ({}));
+      expect(background.backgroundFunction.called).toStrictEqual(false);
+
+      background.connectionStream = {
+        readable: true,
+      };
+      _setBackgroundConnection(background);
+      callBackgroundMethod('backgroundFunction', [], () => ({}));
+      expect(background.backgroundFunction.called).toStrictEqual(true);
+    });
+
+    it('resolves if backgroundFunction called resolves', async () => {
+      const background = {
+        connectionStream: {
+          readable: true,
+        },
+        backgroundFunction: (cb) => {
+          return cb(null, 'test');
+        },
+      };
+      _setBackgroundConnection(background);
+      const mockFn = sinon.stub();
+      await new Promise((resolve) => {
+        callBackgroundMethod('backgroundFunction', [], () => {
+          mockFn();
+          resolve('test');
+        });
+      });
+      expect(mockFn.called).toStrictEqual(true);
+    });
+
+    // This one still needs fixing
+    // it('rejects if backgroundFunction called throws exception', async () => {
+    //   const background = {
+    //     connectionStream: {
+    //       readable: true,
+    //     },
+    //     backgroundFunction: async () => {
+    //       await Promise.resolve('test');
+    //     },
+    //   };
+    //   _setBackgroundConnection(background);
+    //   const mockFn = sinon.stub();
+    //   await new Promise((_, reject) => {
+    //     callBackgroundMethod('backgroundFunction', [], () => {
+    //       reject();
+    //     });
+    //   });
+    //   expect(mockFn.called).toStrictEqual(true);
+    // });
+  });
```

Comment:
Since `mockFn` is a callback with is run for both success and failure I think the last 2 test cases are same and this one is redundant.

Reply:
The other test (now removed) should be covering a failure case. It was still messed up because I had to go AFK before I managed to get Jest to check for a failure. I can't say I like Jest ;)

Reply:
In this case we have single callback for both success / failure. We might still be able to test scenario when exception is thrown.